From 9d1bee3be4fd76433eb9dd219bf23873a87cd67f Mon Sep 17 00:00:00 2001
From: Srujana Challa <schalla@marvell.com>
Date: Thu, 3 Jun 2021 12:57:18 +0530
Subject: [PATCH 1580/1921] crypto: octeontx2: fix CPT block address for
 getting engine capabilities

Engine capabilities are being requested from BLKADDR_CPT1 instead of
BLKADDR_CPT0, so this patch changes the block address to BLKADDR_CPT0.

Fixes 'commit ac49bdb742cb ("crypto: octeontx2: update inline ipsec
mbox to support 98xx")'

Signed-off-by: Srujana Challa <schalla@marvell.com>
Change-Id: Ib8c3588f2ee4440958140a1b27c394fe19ca5f80
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/53388
Reviewed-by: Nithin Kumar Dabilpuram <ndabilpuram@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <sgoutham@marvell.com>
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/crypto/marvell/octeontx2/otx2_cptpf_ucode.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/crypto/marvell/octeontx2/otx2_cptpf_ucode.c b/drivers/crypto/marvell/octeontx2/otx2_cptpf_ucode.c
index 5588fad1e5a4..c1f12a46be02 100644
--- a/drivers/crypto/marvell/octeontx2/otx2_cptpf_ucode.c
+++ b/drivers/crypto/marvell/octeontx2/otx2_cptpf_ucode.c
@@ -1416,7 +1416,7 @@ int otx2_cpt_discover_eng_capabilities(struct otx2_cptpf_dev *cptpf)
 		goto delete_grps;
 
 	otx2_cptlf_set_dev_info(&cptpf->lfs, cptpf->pdev, cptpf->reg_base,
-				&cptpf->afpf_mbox, BLKADDR_CPT1);
+				&cptpf->afpf_mbox, BLKADDR_CPT0);
 	ret = otx2_cptlf_init(&cptpf->lfs, OTX2_CPT_ALL_ENG_GRPS_MASK,
 			      OTX2_CPT_QUEUE_HI_PRIO, 1);
 	if (ret)
-- 
2.31.1

